airflow-core/tests/unit/api_fastapi/core_api/routes/public/test_dags.py

```diff
@@ -220,10 +220,10 @@ class TestGetDags(TestDagEndpoint):
                     "dag_run_state": "failed",
                     "exclude_stale": False,
                 },
-                1,
-                [DAG3_ID],
+                0,
+                [],
```

Comment:
Why does this change?

Reply:
+1

Reply:
The reason is that, dag_run_state behavior was a bit nonsensical, as I mentioned on slack.  I will be removing dag_run_state in followup so this is moot.

But the answer is, it meant "list me the dags that have _any_ dag run in state X".  The reason it changed in the test, is for now it means same as last dag run state.

Reply:
Also ideally we would want a test that creates the join explosion to assert that it's not happening again. (something with multiple dag_run with the same start_date)